### PR TITLE
[Text Generation] Fail if `prompt_sequence_length % 4 != 0`

### PR DESCRIPTION
**Pull request by dbogunowicz**

### CODE REVIEW

File: src/deepsparse/transformers/pipelines/text_generation.py [hunk 1]
```diff
@@ -252,6 +252,15 @@ def __init__(
             if "WAND_OPT_FLAGS" not in os.environ:
                 os.environ["WAND_OPT_FLAGS"] = "default,~pyramids"
 
+        # the current requirement on the deepsparse engine
+        # is that prompt_sequence_length
+        # must be a multiple of four.
+        # for simplicity let's extend this requirement to all engines
+        if prompt_sequence_length % 4 != 0:
```
Comment (mgoin) [inline src/deepsparse/transformers/pipelines/text_generation.py:259]: what about 1 🥇

Comment (rahul-tuli) [reply src/deepsparse/transformers/pipelines/text_generation.py:259]: Michael you got that eagle eye

Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py:259]: woops

Comment (dbogunowicz) [inline src/deepsparse/transformers/pipelines/text_generation.py:259]:
```suggestion
        if prompt_sequence_length % 4 == 0:
```

Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py:259]: thanks for erroneous recommendation copilot...

Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py:259]: im confused. "if x modulo 4 does not equal 0: raise an error" -> this makes sense

Comment (mgoin) [reply src/deepsparse/transformers/pipelines/text_generation.py:259]: we also need to allow prompt_sequence_length=1 for the single-token case so `if prompt_sequence_length is 1 or prompt_sequence_length % 4 != 0:`

Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation.py:259]: oh yeah, haha, silly.

**Review by mgoin — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**